channels/generic/http.py

```diff
@@ -87,3 +73,22 @@ class AsyncHttpConsumer(AsyncConsumer):
         from here.
         """
         pass
+    
+    async def http_request(self, message):
+        """
+        Async entrypoint - concatenates body fragments and hands off control
+        to ``self.handle`` when the body has been completely received.
+        """
+        try:
+            self.body.append(message["body"])
+            if not message.get("more_body"):
+                await self.handle(b"".join(self.body))
+        finally:
+            await self.disconnect()
```

Comment:
This is too early. You don't want to disconnect after the first body chunk already. Maybe only run `await self.disconnect()` if an exception happens? (Not sure about this.) Also, I suspect you should raise `StopConsumer` here as well, same as in `http_disconnect`

Reply:
... to clarify, the original code uses a while loop to receive one or more `http.request` messages. The code you're proposing will only accept one `http.request` message. If the `more_body` flag is set in the first `http.request` message, `handle()` will never be called. That's bad.

Reply:
Yep, I forgot that `finally` would be executed regardless of success. Would this work instead?
```python
        try:
            self.body.append(message["body"])
            if not message.get("more_body"):
                await self.handle(b"".join(self.body))
                
                await self.disconnect()
                raise StopConsumer()
        except:
            await self.disconnect()
            raise StopConsumer()
```

Reply:
Yes, I think that's closer to what works.

Two points: 

- The body is optional [according to the spec](https://asgi.readthedocs.io/en/latest/specs/www.html#request), so you should probably leave check in there whether the "body" is set at all.
- Bare `except:` statements are frowned upon. At least use `except Exception` to not catch SystemExit, KeyboardInterrupt etc. (https://docs.python.org/3/library/exceptions.html#exception-hierarchy)

Reply:
Is it sufficient to do the following to check or are there any falsy values that "body" can have?
```python
body = message.get("body")
if body:
    self.body.append(body)
```